src/gui/TrackView.cpp

```diff
@@ -0,0 +1,473 @@
+/*
+ * TrackView.cpp - implementation of TrackView class
+ *
+ * Copyright (c) 2004-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include "TrackView.h"
+
+#include <assert.h>
+#include <cstdlib>
+
+#include <QLayout>
+#include <QLinearGradient>
+#include <QMenu>
+#include <QMouseEvent>
+#include <QPainter>
+#include <QStyleOption>
+#include <QVariant>
+#include <QClipboard>
+
+
+#include "AutomationPattern.h"
+#include "AutomationTrack.h"
+#include "AutomationEditor.h"
+#include "BBEditor.h"
+#include "BBTrack.h"
+#include "BBTrackContainer.h"
+#include "ConfigManager.h"
+#include "Clipboard.h"
+#include "ColorChooser.h"
+#include "embed.h"
+#include "Engine.h"
+#include "GuiApplication.h"
+#include "FxMixerView.h"
+#include "gui_templates.h"
+#include "MainWindow.h"
+#include "Mixer.h"
+#include "ProjectJournal.h"
```

Comment:
All until (and including) SongEditor.h are not used

Reply:
Mixer is actually needed, as seen in ```TrackView::dropEvent()```.

Reply:
Good point :+1: